File: C/Grace.c
#include <fcntl.h>
#include <unistd.h>
#include <stdio.h>

/*
** dr_quine [grace], 42 fremont, nov 2018
*/

#define GRACE_KID "Grace_kid.c"
#define SELF "#include <fcntl.h>%c#include <unistd.h>%c#include <stdio.h>%c%c/*%c** dr_quine [grace], 42 fremont, nov 2018%c*/%c%c#define GRACE_KID %c%s%c%c#define SELF %c%s%c%c#define FT(x)int main() { int fd; if ((fd = open(x, O_WRONLY | O_CREAT | O_TRUNC, 0644)) > 0) { (void)dprintf(fd,SELF,10,10,10,10,10,10,10,10,34,GRACE_KID,34,10,34,SELF,34,10,10,10,10); (void)close(fd); } return 0; }%c%cFT(GRACE_KID)%c"
#define FT(x)int main() { int fd; if ((fd = open(x, O_WRONLY | O_CREAT | O_TRUNC, 0644)) > 0) { (void)dprintf(fd,SELF,10,10,10,10,10,10,10,10,34,GRACE_KID,34,10,34,SELF,34,10,10,10,10); (void)close(fd); } return 0; }

FT(GRACE_KID)

File: Python/Grace.py
#!/usr/bin/env python3

import sys

#
# amazing grace
#

FT = exec
SELF = '#!/usr/bin/env python3%c%cimport sys%c%c#%c# amazing grace%c#%c%cFT = exec%cSELF = %c%s%c%cGRACE = %cf = open(%cGrace_kid.py%c, %cw%c); sys.stdout=f; print(SELF %% (10,10,10,10,10,10,10,10,10,39,SELF,39,10,39,34,34,34,34,39,10,10)); f.close();%c%c%cFT(GRACE)'
GRACE = 'f = open("Grace_kid.py", "w"); sys.stdout=f; print(SELF % (10,10,10,10,10,10,10,10,10,39,SELF,39,10,39,34,34,34,34,39,10,10)); f.close();'

FT(GRACE)
